sklearn/utils/validation.py

```diff
@@ -203,8 +203,8 @@ def _num_samples(x):
 
     try:
         return len(x)
-    except TypeError:
-        raise TypeError(message)
+    except TypeError as type_error:
+        raise TypeError(message) from type_error
```

Comment:
Surely this should just be bare `raise`?

Reply:
don't we still want the `message` to be passed?

Reply:
I'm with @NicolasHug here, the message is valuable.

Reply:
@jnothman Nudge

Reply:
I think the argument for keeping the bare `raise` here, is that the additional message may case too much noise?

Reply:
We have the error message in master, I see no reason to remove it now.

I'll merge the PR, @jnothman please feel free to raise any concern if any